modules/core/src/main/java/org/apache/synapse/config/xml/endpoints/AddressEndpointFactory.java

```diff
@@ -80,6 +81,17 @@ protected Endpoint createEndpoint(OMElement epConfig, boolean anonymousEndpoint,
             addressEndpoint.setName(name.getAttributeValue());
         }
 
+        //***
+        OMElement httpStatusElement = epConfig.getFirstChildWithName(new QName(SynapseConstants.SYNAPSE_NAMESPACE, "httpStatus"));
+        if(httpStatusElement != null) {
+            ArrayList<String> l = new ArrayList<String>();
+            l.add("500");
```

Comment:
I used hard coded values because I'm still working on UI changes to add failover http status codes.